Store.MVC/Controllers/CakeController.cs

```diff
@@ -18,18 +20,32 @@ public CakeController(ICakeRepository repository)
             cakeRepository = repository;
         }
 
-        public ViewResult List(int page = 1)
-        {   
-
-            return View(cakeRepository.Cakes
-                   .OrderBy(c => c.Id)
-                   .Skip((page - 1)*pageSize)
-                   .Take(pageSize));
-        }
+        public ViewResult List(string category, int page = 1)
+        {
+            CakeListView model = new CakeListView
+            {
+                Cakes = cakeRepository.Cakes
+                   .Where(c => category == null || c.Category == category)
```

Comment:
As I mentioned above - create a service including repository as dependency and then move all LINQ methods to it